Karl/test_service/filter.h

```diff
@@ -55,41 +55,23 @@ class ServiceFilter final {
     // In example "production" mode just start regular keepalives.
     claire_.Register();
 #endif
-  }
-
-  ~ServiceFilter() {
-    destructing_ = true;
-    thread_.join();
+    claire_.AddDependency(service_annotated);
```

Comment:
Isn't this initialized by an initializer list in Claire's construction in the initialization list?